homeassistant/components/eight_sleep/__init__.py

```diff
@@ -168,9 +189,27 @@ def __init__(
 
         mapped_name = NAME_MAP.get(sensor, sensor.replace("_", " ").title())
         if self._user_obj is not None:
-            mapped_name = f"{self._user_obj.side.title()} {mapped_name}"
-
-        self._attr_name = f"Eight {mapped_name}"
+            self._attr_name = (
+                f"{self._user_obj.user_profile['firstName']}'s {mapped_name}"
+            )
+        else:
+            self._attr_name = f"Eight Sleep {mapped_name}"
         self._attr_unique_id = (
             f"{_get_device_unique_id(eight, self._user_obj)}.{sensor}"
         )
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, _get_device_unique_id(eight, self._user_obj))}
+        )
```

Comment:
It would be nice to fill this out a bit more.  I think you at least have the `name`

Reply:
This pre-creates the device info, this logic here is just to connect the entity to that device:
https://github.com/home-assistant/core/pull/71095/files#diff-e9fa84532f08b51681ede6b51a04754c32f89ec8932ef6a568525e48ed7c7e18R134-R139